akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/PersistentBehavior.scala

```diff
@@ -58,43 +59,38 @@ abstract class PersistentBehavior[Command, Event, State >: Null](val persistence
   protected def eventHandler(): EventHandler[Event, State]
 
   /**
-   * @return A new, mutable, by state command handler builder
+   * @param stateClass The handlers defined by this builder are used when the state is an instance of the `stateClass`
+   * @return A new, mutable, command handler builder
    */
-  protected final def commandHandlerBuilder(): CommandHandlerBuilder[Command, Event, State] =
-    new CommandHandlerBuilder[Command, Event, State]()
+  protected final def commandHandlerBuilder[S <: State](stateClass: Class[S]): CommandHandlerBuilder[Command, Event, S, State] =
+    CommandHandlerBuilder.builder[Command, Event, S, State](stateClass)
```

Comment:
I tried to move the CommandHandlerBuilder to an inner class of PersistentBehavior to reduce number of type parameters, but that was the reason for the problem I showed at the meeting.

Reply:
I double checked the type bound here, because worried about the same issue as streams, but the signature renders to:

```
public final <S extends State> akka.persistence.typed.javadsl.CommandHandlerBuilder<Command, Event, S, State> commandHandlerBuilder(java.lang.Class<S>);
```

So in this case it is safe, the problem only is when there is a lower bound.